saleor/discount/utils.py

```diff
@@ -186,6 +187,81 @@ def get_voucher_code_instance(
     return code_instance
 
 
+def apply_voucher_to_line(
+    voucher_info: "VoucherInfo",
+    lines_info: Iterable["LineInfo"],
+):
+    """Attach voucher to valid checkout or order lines info.
+
+    Apply a voucher to checkout/order line info when the voucher has the type
+    SPECIFIC_PRODUCTS or is applied only to the cheapest item.
+    """
+    from .utils import get_discounted_lines
```

Comment:
Do you still need this? `get_discounted_lines` is placed below this function.

Reply:
Fixed